### DISCUSSION TITLE
Source Bank - Target Bank

### DISCUSSION THREAD

**Discussion by georgelza** [brianvoe/gofakeit#229] [Ideas]
Hi there
How would you suggest I create fake source bank (originating) and target (destination) bank information.
I have like 20 Bank names which should be used and then a format of the bank account number and 
Ie create {
Bank Name: xxxxxxx     - want this to be seeded out of my input seed list
Branch Name :yyyyyy - Want this to be seeded out of a input/seed list
AccountNumber: xxxx-xxxxxxxxxx    -  (would be great if i was able to provide multiple structures tied to the bank name/option selected) 
}

G

**Comment by brianvoe** [comment]
The way i would approach it is adding any lookup functions that you need. Both adding data and adding lookup functions have example usages. 
https://github.com/brianvoe/gofakeit/blob/master/data/data_test.go#L69
https://github.com/brianvoe/gofakeit/blob/master/lookup_test.go#L11

once you have those in you can run struct generation with your function name as the fake tag.

**Reply by georgelza** [reply]
... guessing if i have a input set of 20 i could use a random number generator to random select a value 0->20 and then extract that bank name from a array and similar on branch locations, then utilise your random account generation capability, or build up a account number by again random generating numbers of a length between a starting number and end, and then concatenate them together...
Just through originating and destination bank accounts would be a regular ask from a transaction view point, similar to generating credit cards details, thanking FakeIt might have had the ability already vs me writing it ;)... aka me=lazy don't do something if already done by someone else ;)

**Reply by brianvoe** [reply]
If you have a dataset i can add and it makes sense to. Ill add it.

**Reply by georgelza** [reply]
My thinking would be a input array as a seed set to a list of banks, similar a input array seed set to a list of branches.
This model can actually be utilised on various of the fake data generation as to make the generated data more relevant to the user. 2cent idea... ;)

**Reply by brianvoe** [reply]
If you have a dataset i can add and it makes sense to. Ill add it.
